src/main/java/teammates/common/util/TzdbResourceZoneRulesProvider.java

```diff
@@ -142,4 +151,31 @@ private void load(DataInputStream dis) throws Exception {
     public String toString() {
         return "TZDB[" + versionId + "]";
     }
+
+    /**
+     * Modified from {@link java.time.zone.Ser#read}.
+     */
+    private static Object serRead(DataInput in) throws IOException {
+        byte type = in.readByte();
+        switch (type) {
+        case ZRULES:
+            return invokeReadExternal(ZoneRules.class, in); // ZoneRules.readExternal(in)
+        case ZOT:
+            return invokeReadExternal(ZoneOffsetTransition.class, in); // ZoneOffsetTransition.readExternal(in)
+        case ZOTRULE:
+            return invokeReadExternal(ZoneOffsetTransitionRule.class, in); // ZoneOffsetTransitionRule.readExternal(in)
+        default:
+            throw new StreamCorruptedException("Unknown serialized type");
+        }
+    }
+
+    private static Object invokeReadExternal(Class<?> cls, DataInput in) throws IOException {
```

Comment:
Sorry if this is a dump question, but why do we need to use reflection here?

Reply:
Those methods are not public :)

Noted on the documentation request :+1: